noir/compiler/noirc_frontend/src/hir/def_collector/dc_crate.rs

```diff
@@ -256,6 +256,14 @@ impl DefCollector {
         // Add the current crate to the collection of DefMaps
         context.def_maps.insert(crate_id, def_collector.def_map);
 
+        for macro_processor in &macro_processors {
```

Comment:
Why did we need to add this?

Reply:
Do you mean the borrow, or the new function?

The new function exists because we want to be able to inject code before resolution (meaning we can inject ast-like nodes, created by parsing raw code), but after some collection has been done (critically, collection over dependencies). The two existing functions are called either too soon or too late.

The borrow happens because `macro_processors` is later cloned, so we can't move it here. That clone is however meaningless since the macro structs, hold no data, there's nothing to clone. But addressing that involved a larger refactor of how macros are used that I didn't want to include here.

Reply:
Ideally we should be able to remove `process_untyped_ast` for a new `process_collected_crate` method or something that accepts a mutable reference to the entire def collector and context. Right now this more specific macro method is sufficient until we can refactor that though.

Reply:
See https://github.com/AztecProtocol/aztec-packages/issues/4653